### PR TITLE
♨️ Properly update error when switching to not configured network

### PR DESCRIPTION
**Pull request by jakvbs**
Closes https://github.com/TrueFiEng/useDApp/issues/533

### CODE REVIEW

File: packages/core/src/hooks/useEthers.ts [hunk 1]
```diff
@@ -64,14 +65,25 @@ export function useEthers(): Web3Ethers {
     isLoading,
   } = useNetwork()
 
-  const { networks } = useConfig()
+  const { networks, readOnlyUrls } = useConfig()
+  const error = useRef<Error | undefined>()
```
Comment (rzadp) [inline packages/core/src/hooks/useEthers.ts:69]: Why `useRef`? Is it needed? Why not just `useState`?

Comment (jakvbs) [reply packages/core/src/hooks/useEthers.ts:69]: Ah yes, I forgot to change it

File: packages/core/src/hooks/useEthers.test.tsx [hunk 2]
```diff
@@ -58,6 +64,25 @@ describe('useEthers', () => {
...
+        }, [])
+
+        return useEthers()
+      },
+      { config }
+    )
+
+    await waitForCurrent((val) => !!val.error)
+    expect(result.current.error).not.to.be.undefined
+    expect(result.current.error?.toString()).to.include(`Not configured chain id: ${Localhost.chainId}`)
```
Comment (rzadp) [inline packages/core/src/hooks/useEthers.test.tsx:83]: Can we have test for both not configured and not supported chains?

Comment (jakvbs) [reply packages/core/src/hooks/useEthers.test.tsx:83]: There are different error messages for both cases, but we can only check if the error occured. Should we?

Comment (rzadp) [reply packages/core/src/hooks/useEthers.test.tsx:83]: I meant 2 tests

**Pull request merged**